### PR TITLE
Initial test framework

### PR DESCRIPTION
**Pull request by azabbasi**
There is quite a bit of difference between azure test core for Java and azure test core for .NET
One of them being that Test framework for java doesn't support encrypted user configuration.

For the time being we have to store the configurations as environment variables until I hash out a path forward to make that a lot more convenient.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -9,6 +9,9 @@ temp/
 # Spring
 **/*packageOutputDirectory*
 
+# Sensitive files
```
Comment (azabbasi) [inline .gitignore:12]: This is required to make sure no one checks in their encrypted environment variables.
There is an identical rule for .NET repo

**Comment by drwill-ms** [comment]
> There is quite a bit of difference between azure test core for Java and azure test core for .NET
> One of them being that Test framework for java doesn't support encrypted user configuration.
> 
> For the time being we have to store the configurations as environment variables until I hash out a path forward to make that a lot more convenient.

Have you reached out to Pavel and Srikanta to ask about feature parity in Java for encrypted user config?

File: .gitignore (hunk 1 above)
Comment (drwill-ms) [reply .gitignore:12]: Do you really want to add this to the root of the repo?

Comment (drwill-ms) [reply .gitignore:12]: I guess so if you are copying what was done in the net repo

File: sdk/digitaltwins/azure-digitaltwins-core/src/test/java/com/azure/digitaltwins/core/DigitalTwinsTestBase.java [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+    protected static final String CLIENT_SECRET = Configuration.getGlobalConfiguration()
+        .get("CLIENT_SECRET", "clientSecret");
+
+    protected static final String CLIENT_ID = Configuration.getGlobalConfiguration()
+        .get("CLIENT_ID", "clientId");
+
+    protected static final String DIGITALTWINS_URL = Configuration.getGlobalConfiguration()
+        .get("DIGITALTWINS_URL", "https://playback.api.wus2.digitaltwins.azure.net");
+
+    protected DigitalTwinsClientBuilder getDigitalTwinsClientBuilder(){
```
Comment (drwill-ms) [inline sdk/digitaltwins/azure-digitaltwins-core/src/test/java/com/azure/digitaltwins/core/DigitalTwinsTestBase.java:28]: space between `()` and `{`?

File: sdk/digitaltwins/azure-digitaltwins-core/src/test/java/com/azure/digitaltwins/core/DigitalTwinsTestBase.java [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+        DigitalTwinsClientBuilder builder = new DigitalTwinsClientBuilder()
+            .endpoint(DIGITALTWINS_URL)
+            .tokenCredential(new ClientSecretCredentialBuilder()
+                .tenantId(TENANT_ID)
+                .clientId(CLIENT_ID)
+                .clientSecret(CLIENT_SECRET)
+                .build());
+
+        if(interceptorManager.isPlaybackMode()){
```
Comment (drwill-ms) [inline sdk/digitaltwins/azure-digitaltwins-core/src/test/java/com/azure/digitaltwins/core/DigitalTwinsTestBase.java:37]: space between `if` and `(`?

File: sdk/digitaltwins/azure-digitaltwins-core/src/test/java/com/azure/digitaltwins/core/DigitalTwinsTestBase.java [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+        // If it is record mode, we add record mode policies to the builder.
+        // There is no isRecordMode method on interceptorManger.
+        if(!interceptorManager.isLiveMode()){
+            builder.addPolicy(interceptorManager.getRecordPolicy());
+        }
+
+        return builder;
+    }
+
+    protected DigitalTwinsClientBuilder getDigitalTwinsClientBuilder(HttpPipelinePolicy... policies){
```
Comment (abhipsaMisra) [inline sdk/digitaltwins/azure-digitaltwins-core/src/test/java/com/azure/digitaltwins/core/DigitalTwinsTestBase.java:53]: nice!

File: sdk/digitaltwins/azure-digitaltwins-core/src/test/prerequisite/manifest.json [hunk 5]
```diff
@@ -0,0 +1,9 @@
+[{
+    "resourceAppId": "0b07f429-9f4b-4714-9392-cc5e8e80c8b0",
```
Comment (drwill-ms) [inline sdk/digitaltwins/azure-digitaltwins-core/src/test/prerequisite/manifest.json:2]: what are these values?

Comment (azabbasi) [reply sdk/digitaltwins/azure-digitaltwins-core/src/test/prerequisite/manifest.json]: this the the dataplane APP ID for ADT

**Comment by azabbasi** [comment]
> > There is quite a bit of difference between azure test core for Java and azure test core for .NET
> > One of them being that Test framework for java doesn't support encrypted user configuration.
> > For the time being we have to store the configurations as environment variables until I hash out a path forward to make that a lot more convenient.
> 
> Have you reached out to Pavel and Srikanta to ask about feature parity in Java for encrypted user config?

I talked to someone on the JAVA SDK team and he encouraged me to open an issue against the test framework. In the meanwhile, I will start a conversation between myself, Pavel and Srikanta to raise this issue.

File: sdk/digitaltwins/azure-digitaltwins-core/src/test/prerequisite/manifest.json (hunk 5 above)
Comment (drwill-ms) [reply sdk/digitaltwins/azure-digitaltwins-core/src/test/prerequisite/manifest.json]: Why not use the human readable one: https://digitaltwins.azure.net?

**Comment by drwill-ms** [comment]
> > > There is quite a bit of difference between azure test core for Java and azure test core for .NET
> > > One of them being that Test framework for java doesn't support encrypted user configuration.
> > > For the time being we have to store the configurations as environment variables until I hash out a path forward to make that a lot more convenient.
> > 
> > 
> > Have you reached out to Pavel and Srikanta to ask about feature parity in Java for encrypted user config?
> 
> I talked to someone on the JAVA SDK team and he encouraged me to open an issue against the test framework. In the meanwhile, I will start a conversation between myself, Pavel and Srikanta to raise this issue.

I guess my bigger question is, when we get a feature in one language repo, is it automatically in the backlog for other languages, or do we have to ask for it explicitly?

File: .gitignore (hunk 1 above)
Comment (azabbasi) [reply .gitignore:12]: yes :)

File: sdk/digitaltwins/azure-digitaltwins-core/src/test/prerequisite/manifest.json (hunk 5 above)
Comment (azabbasi) [reply sdk/digitaltwins/azure-digitaltwins-core/src/test/prerequisite/manifest.json]: we no longer need this anyways, removing.

File: sdk/digitaltwins/azure-digitaltwins-core/pom.xml [hunk 6]
```diff
@@ -15,28 +15,65 @@
     <artifactId>azure-digitaltwins-core</artifactId>
     <version>1.0.0-beta.1</version> <!-- {x-version-update;com.azure:azure-digitaltwins-core;current} -->
 
+  <properties>
+    <!-- These are the code coverage settings that should eventually be removed to inherit them from the parent pom file. for the time being we have a very low code coverage so we are dropping it to a very low setting -->
```
Comment (bikamani) [inline sdk/digitaltwins/azure-digitaltwins-core/pom.xml:19]: 👍

**Pull request merged**